consensus/spos/bls/sideChainSubroundBlock.go

```diff
@@ -0,0 +1,74 @@
+package bls
+
+import (
+	"github.com/ElrondNetwork/elrond-go/common"
+	"github.com/ElrondNetwork/elrond-go/consensus"
+	"github.com/ElrondNetwork/elrond-go/consensus/spos"
+	"golang.org/x/net/context"
+	"time"
+)
+
+type sideChainSubroundBlock struct {
+	*subroundBlock
+}
+
+// NewSideChainSubroundBlock creates a sideChainSubroundBlock object
+func NewSideChainSubroundBlock(subroundBlock *subroundBlock) (*sideChainSubroundBlock, error) {
```

Comment:
you are still missing a constructor for the consensus - the one which creates all the subrounds. You want to do that on the other repo ? also the name should suggest something else. As this can be later used here as well.
Rename to something better the struct and the constructor.

Reply:
Done

Reply:
I just kept the name of the original struct / constructor and prefixed them with SideChain. This was the approach also used for the other components adapted for side chain like ForkDetector and BlockTracker

Reply:
you did not change the name. We need to add SideChain only which is specific to that one. This is a general consensus optimization.

Reply:
done